.evergreen/.evg.yml

```diff
@@ -1813,15 +1833,22 @@ buildvariants:
     - name: "plain-auth-test"
 
 - matrix_name: "aws-auth-test"
-  matrix_spec: { ssl: "nossl", jdk: ["jdk8", "jdk17"], version: ["4.4", "5.0", "6.0", "latest"], os: "ubuntu" }
-  display_name: "MONGODB-AWS Auth test ${version} ${jdk}"
+  matrix_spec: { ssl: "nossl", jdk: ["jdk8", "jdk17"], version: ["4.4", "5.0", "6.0", "latest"], os: "ubuntu",
+                 aws-credential-provider: "*" }
+  display_name: "MONGODB-AWS Auth test ${version} ${jdk} ${aws-credential-provider}"
```

Comment:
Do we need to run all tests on all variants? 

Do we need to setup any AWS specific environment variables to test the v2/v1 sdks pick them up?

Reply:
> Do we need to run all tests on all variants?

You could argue that as long as we can prove that _any_ credential fetching is working for the SDK case, then _all_ of them will work.  @blink1073 suggested that we test all cases, so I went with that just to be safe.  But it might be too safe.  

One possibility is that we test all the different credential-fetching scenarios in a single task (one task for each of the three fetching implementations).  That will reduce the number of tasks 

> Do we need to setup any AWS specific environment variables to test the v2/v1 sdks pick them up?

No, the SDK uses the same environment variables as we do.  Our built-in implementation is intended to emulate the SDK.

Reply:
We run the Python AWS auth tests against MongoDB 4.4, 5.0, 6.0, rapid, and latest.  We run against a single version of Python on macos and ubuntu, and all supported versions of Python on Windows.  The ECS and EC2 tests only run on Ubuntu.

Reply:
Thanks - resolving.